crates/egui/src/text_selection/visuals.rs

```diff
@@ -96,8 +96,17 @@ pub fn paint_text_selection(
 pub fn paint_cursor_end(painter: &Painter, visuals: &Visuals, cursor_rect: Rect) {
     let stroke = visuals.text_cursor.stroke;
 
-    let top = cursor_rect.center_top();
-    let bottom = cursor_rect.center_bottom();
+    let (top, bottom) = if (stroke.width as usize) % 2 == 0 {
+        (
+            painter.round_pos_to_pixels(cursor_rect.center_top()),
+            painter.round_pos_to_pixels(cursor_rect.center_bottom()),
+        )
+    } else {
+        (
+            painter.round_pos_to_pixel_center(cursor_rect.center_top()),
+            painter.round_pos_to_pixel_center(cursor_rect.center_bottom()),
+        )
+    };
```

Comment:
This could use a comment explaining what is going on, and ideally with a link to https://github.com/emilk/egui/issues/5164 so we can remember to update this code if we move the rounding to epaint

Reply:
Comment added!